script_test.go

```diff
@@ -1850,6 +1850,22 @@ func TestReadReturnsErrorGivenReadErrorOnPipe(t *testing.T) {
 	}
 }
 
+func TestWaitForErrorOnPipe(t *testing.T) {
+	t.Parallel()
+	p := script.Echo("a\nb\nc\n").ExecForEach("{{invalid template syntax}}")
+	if p.Wait() == nil {
+		t.Error("want error, got nil")
+	}
+}
+
+func TestWaitForNoErrorOnPipe(t *testing.T) {
+	t.Parallel()
+	p := script.Echo("a\nb\nc\n").ExecForEach("echo \"{{.}}\"")
+	if err := p.Wait(); err != nil {
```

Comment:
I would expect the test to not only check for the error, but also to validate the error type.

I'm talking about checking the error type with errors.Is

Sometimes a refactoring could change the behavior and an unexpected error will be returned. But then the tests won't see it, so users will face it later when merged

Reply:
I'm not entirely sure what value we get from checking the specific error type. The goal of this test is to just test that if there is an error present on the pipe, we can get it using `Wait`.

Reply:
It's true that you are replicating an existing test.

I would have expected that both are catching the expected error for an invalid template.

The fact, I wasn't able to convince you earlier, let me think either I cannot convince you, or that I'm not able to explain it.

I don't mind if you keep the current code, it's a test. It was also a side remark.

Reply:
> I would expect the test to not only check for the error, but also to validate the error type.

That's not a crazy idea, but it's not something I generally do unless returning some _specific_ error type is an important part of the function's behaviour. If, on the other hand, as is usually the case, all that matters is that it returns _some_ error, then checking it's not `nil` is good enough.

I think it's wise not to test more than you really need to, for fear of ending up with overly fragile tests.

Reply:
Indeed. I understand your point